### PR TITLE
Comment on MementoService::put

### PR DESCRIPTION
**Pull request by ajs6f**
Meets https://github.com/trellis-ldp/trellis/issues/346

### CODE REVIEW

File: core/api/src/main/java/org/trellisldp/api/MementoService.java [hunk 1]
```diff
@@ -32,10 +32,13 @@
...
-     * @apiNote the default implementation of this method fetches a resource from a {@link ResourceService} that is
-     *          external to the Memento service. In the case that the two services are managed by the same persistence
-     *          layer, it may not be necessary to fetch a {@link Resource} from the persistence layer, in which case
-     *          this method can be overridden as a no-op method, e.g. {@code return completedFuture(null);}.
+     * @implSpec The default implementation of this method fetches a resource from a {@link ResourceService} that is
+     * external to the Memento service. 
+     * @implNotes In the case that the two services are managed by the same persistence layer, it may not be
+     * necessary to fetch a {@link Resource} from the persistence layer, in which case this method can be
+     * overridden as a no-op method, e.g. {@code return completedFuture(null);}.
+     * @apiSpec An implementation may choose to store a new Memento only when this method is called,
```
Comment (acoburn) [inline core/api/src/main/java/org/trellisldp/api/MementoService.java]: There isn't a defined `@apiSpec` tag -- only `@apiNote`, `@implNote` and `@implSpec`.

File: core/api/src/main/java/org/trellisldp/api/MementoService.java [hunk 2]
```diff
@@ -32,10 +32,13 @@
      * Create a new Memento for a resource, retrieved from a {@link ResourceService}.
      * @param resourceService the resource service.
      * @param identifier the identifier.
-     * @apiNote the default implementation of this method fetches a resource from a {@link ResourceService} that is
-     *          external to the Memento service. In the case that the two services are managed by the same persistence
-     *          layer, it may not be necessary to fetch a {@link Resource} from the persistence layer, in which case
-     *          this method can be overridden as a no-op method, e.g. {@code return completedFuture(null);}.
+     * @implSpec The default implementation of this method fetches a resource from a {@link ResourceService} that is
+     * external to the Memento service. 
+     * @implNotes In the case that the two services are managed by the same persistence layer, it may not be
```
Comment (acoburn) [inline core/api/src/main/java/org/trellisldp/api/MementoService.java]: `@implNote`

File: core/api/src/main/java/org/trellisldp/api/MementoService.java [hunk 3]
```diff
@@ -32,10 +32,13 @@
...
-     *          external to the Memento service. In the case that the two services are managed by the same persistence
-     *          layer, it may not be necessary to fetch a {@link Resource} from the persistence layer, in which case
-     *          this method can be overridden as a no-op method, e.g. {@code return completedFuture(null);}.
+     * @implSpec The default implementation of this method fetches a resource from a {@link ResourceService} that is
+     * external to the Memento service. 
+     * @implNote In the case that the two services are managed by the same persistence layer, it may not be
+     * necessary to fetch a {@link Resource} from the persistence layer, in which case this method can be
+     * overridden as a no-op method, e.g. {@code return completedFuture(null);}.
+     * @implSpec An implementation may choose to store a new Memento only when this method is called,
+     * or at other times as well, e.g. when {@link ResourceService::replace} is called.
```
Comment (acoburn) [inline core/api/src/main/java/org/trellisldp/api/MementoService.java]: `ResourceService::replace` => `ResourceService#replace`

Comment (ajs6f) [reply core/api/src/main/java/org/trellisldp/api/MementoService.java]: Ah, sure. I wish Java would settle on one notation.

File: core/api/src/main/java/org/trellisldp/api/MementoService.java [hunk 4]
```diff
@@ -32,10 +32,13 @@
      * Create a new Memento for a resource, retrieved from a {@link ResourceService}.
      * @param resourceService the resource service.
      * @param identifier the identifier.
-     * @apiNote the default implementation of this method fetches a resource from a {@link ResourceService} that is
-     *          external to the Memento service. In the case that the two services are managed by the same persistence
-     *          layer, it may not be necessary to fetch a {@link Resource} from the persistence layer, in which case
-     *          this method can be overridden as a no-op method, e.g. {@code return completedFuture(null);}.
+     * @implSpec The default implementation of this method fetches a resource from a {@link ResourceService} that is
+     * external to the Memento service.
```
Comment (acoburn) [inline core/api/src/main/java/org/trellisldp/api/MementoService.java]: Looks like a "trailing whitespace" error on this line. Urg.

Comment (ajs6f) [reply core/api/src/main/java/org/trellisldp/api/MementoService.java]: Oh, how I despise Checkstyles.

**Comment by acoburn** [comment]
Thanks!

**Pull request merged**